### PR TITLE
Add support for StrictHostKeyChecking and UserKnownHostsFile

### PR DESCRIPTION
**Pull request by snaggen**
As I realize I use StrictHostKeyChecking and UserKnownHostsFile in some of my .ssh/config files, I have added support for them in russh-config.

### CODE REVIEW

**Comment by snaggen** [comment]
I noticed that id files can be specified as:
```
    IdentityFile '~/.ssh/id_ed25519_sk'
```
So I included support for stripping the surrounding ' characters

File: russh-config/src/lib.rs [hunk 1]
```diff
@@ -167,6 +174,35 @@ pub fn parse(file: &str, host: &str) -> Result<Config, Error> {
                         "ask" => config.add_keys_to_agent = AddKeysToAgent::Ask,
                         _ => config.add_keys_to_agent = AddKeysToAgent::No,
                     },
+                    "userknownhostsfile" => {
+                        let mut id = value.trim_start();
+                        if id.len() > 1 && id.starts_with('\'') && id.ends_with('\'') {
+                            id = &id[1..id.len() - 1];
+                        }
+                        if id.starts_with("~/") {
```
[the remark this answers is not in the document]
Comment (snaggen) [reply russh-config/src/lib.rs:182]: The ~/ expansion was a copy of the identity file parsing. But sure I can fox that also.
And about the quotes, it works fine as I have it in my .ssh/config file... Not sure why I have it though, but it is there and works.
I will make it its own function.

**Comment by snaggen** [comment]
I have broken out the functions (did it as traits on &str to make it nice and clean in the match statement), and I also added some tests to ensure things works as expected

**Comment by Eugeny** [comment]
Thank you! @all-contributors add @snaggen for code

**Pull request merged**